File: leetcode/num199.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        result = []
        if root is None:
            return result
        queue = []
        queue.append(root)
        front = None
        while len(queue) != 0:
            size = len(queue)
            for i in range(size):
                front = queue[0]
                queue.remove(queue[0])
                if front.left != None:
                    queue.append(front.left)
                if front.right != None:
                    queue.append(front.right)
                if i == size-1:
                    result.append(front.val)
        return result
            

        

File: leetcode/num199.java
package leetcode;
/**
 * 二叉树的右视图
 */
import java.util.LinkedList;
import java.util.List;

public class num199 {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> list = new LinkedList<>();
        if(root == null) return list;
        LinkedList<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while(!queue.isEmpty()){
            int size = queue.size();
            while(size-->0){
                TreeNode node = queue.pollFirst();
                if(node.left != null) queue.add(node.left);
                if(node.right != null) queue.add(node.right);
                if(size==0) list.add(node.val);
            }
        }
        return list;
    }
}